Sources/OpenCombine/CurrentValueSubject.swift

```diff
@@ -14,20 +14,33 @@ public final class CurrentValueSubject<Output, Failure: Error>: Subject {
     // TODO: Combine uses bag data structure
     private var _subscriptions: [Conduit] = []
 
+    private var _value: Output
+
     private var _completion: Subscribers.Completion<Failure>?
 
     /// The value wrapped by this subject, published as a new element whenever it changes.
     public var value: Output {
-        didSet {
-            send(value)
+        get {
+            defer {
+                _lock._unlock()
+            }
+            _lock._lock()
+            return _value
+        }
+        set {
+            defer {
+                _lock._unlock()
+            }
+            _lock._lock()
```

Comment:
I see that the synchronization test failure reproduces every time. Have you tried removing this locking?